include/maths/analytics/CBoostedTreeHyperparameters.h

```diff
@@ -749,8 +791,8 @@ class MATHS_ANALYTICS_EXPORT CBoostedTreeHyperparameters {
 
     //@ \name Hyperparameter Optimisation
     //@{
-    bool m_StopHyperparameterOptimizationEarly{true};
-    bool m_StoppedHyperparameterOptimizationEarly{false};
+    bool m_EarlyHyperparameterOptimizationStoppingEnabled{true};
+    mutable bool m_StopHyperparameterOptimizationEarly{false};
```

Comment:
This name seems worse to me. This tells you if we did in fact stop hyperparameter optimisation early. `m_StopHyperparameterOptimizationEarly` is an imperative so suggests we're telling the class to "stop hyperparameter optimization early". If you don't like `m_StoppedHyperparameterOptimizationEarly` maybe `m_DidStopHyperparameterOptimizationEarly` or some such.

Reply:
Let's discuss the appropriate names offline.